runtime/src/consensus/state/staking.rs

```diff
@@ -0,0 +1,72 @@
+use std::collections::BTreeMap;
+
+use anyhow::{anyhow, Result};
+use io_context::Context;
+
+use crate::{
+    common::{
+        cbor,
+        key_format::{KeyFormat, KeyFormatAtom},
+    },
+    consensus::{
+        address::Address,
+        staking::{Account, Delegation},
+    },
+    key_format,
+    storage::mkvs::ImmutableMKVS,
+};
+
+/// Consensus staking state wrapper.
+pub struct ImmutableState<'a, T: ImmutableMKVS> {
+    pub mkvs: &'a T,
+}
+
+key_format!(AccountsKeyFmt, Address, 0x50);
+key_format!(TotalSupplyKeyFmt, (), 0x51);
+key_format!(CommonPoolKeyFmt, (), 0x52);
+key_format!(DelegationKeyFmt, (Address, Address), 0x53);
+key_format!(DebondingDelegationKeyFmt, (Address, Address, u64), 0x54);
+key_format!(DebondingQueueKeyFmt, (u64, Address, Address), 0x55);
+key_format!(ParametersKeyFmt, (), 0x56);
+key_format!(LastBlockFees, (), 0x57);
+key_format!(EpochSigningKeyFmt, (), 0x58);
+key_format!(GovernanceDepositsKeyFmt, (), 0x59);
+
+// TODO: implement the rest. Add some tests?
```

Comment:
Would be cool to have interoperability tests for all the methods. But this can be part of a separate issue/PR.

Reply:
> Agreed. This should utilize the existing protocol server right? Need to look a bit into it, but probably makes sense to do in a separate PR. However I wouldn't trust using these methods until we do have such tests grin

You mean the one used for storage interop tests? Yeah that could work but you would first need to import/load a database that has some consensus state in it.

> Additionally to that in the long run we should probably add a txsource workload that does queries via a runtime and directly - and compares the results.

Good idea, feel free to file issues.

Reply:
> You mean the one used for storage interop tests? Yeah that could work but you would first need to import/load a database that has some consensus state in it.

Yeah i have an outline of this working now. Added a flag to the interop protocol server to initialize some debug consensus state on the go side. Depending on when this PR get's approved i might include the tests in this PR, otherwise i'll open an isuse and a follow up PR.